lib/datadog/core/workers/interval_loop.rb

```diff
@@ -7,6 +7,7 @@ module IntervalLoop
         BACK_OFF_RATIO = 1.2
         BACK_OFF_MAX = 5
         BASE_INTERVAL = 1
+        GLOBAL_MUTEX = Mutex.new
```

Comment:
```suggestion
          # This single shared mutex is used to avoid concurrency issues during the
          # per-instance lazy-initialized mutexes.
          MUTEX_INIT = Mutex.new
```